### PR TITLE
ci: upload KIB release template to cloud director organization 

### PR DESCRIPTION
**Pull request by supershal**
**What problem does this PR solve?**:
- Script to create a temporary VM in production vCenter environment, download given template and upload it to VCD organization
- Cleanup Makefile that creates and destroys bastion VMs for easier local testing

**Which issue(s) does this PR fix?**:

* https://d2iq.atlassian.net/browse/D2IQ-97576

**Special notes for your reviewer**:

Test CI job: https://github.com/mesosphere/konvoy-image-builder/actions/runs/5522031816
This PR will allow to invoke this flow manually from GHA UI.
I will file separate PR to kick off this flow when vSphere template is released by the release job.

### CODE REVIEW

File: test/infra/Makefile [hunk 1]
```diff
@@ -2,14 +2,13 @@
 # ---------------------------------------------------------------------
 export GOOS ?= $(shell go env GOOS 2>/dev/null)
 export GOARCH ?= $(shell go env GOARCH 2>/dev/null)
-export TERRAFORM_VERSION ?= 0.15.3
+export TERRAFORM_VERSION ?= 1.5.2
```
Comment (faiq) [inline test/infra/Makefile:5]: Does bumping this have side effects for any existing automation?

**Review by faiq — APPROVED** [review]

File: test/infra/Makefile (hunk 1 above)
Comment (supershal) [reply test/infra/Makefile:5]: Good point. let me run test on all the providers.

**Comment by supershal** [comment]
added https://github.com/mesosphere/konvoy-image-builder/pull/855/commits/c9055fcd2045bc008ca5b636b580f2b4db3b3d40 to call vcloud director upload job when a vsphere template release job is completed.

**Comment by supershal** [comment]
CI job to upload templates to VCD: https://github.com/mesosphere/konvoy-image-builder/actions/runs/5581795415

**Comment by supershal** [comment]
- modified GHA jobs to use composite actions so that we can upload each OS template to VCD.
https://github.com/mesosphere/konvoy-image-builder/actions/runs/5583156285/

- ran e2e tests to verify existing tests are not affected by code changes.

**Comment by dkoshkin** [comment]
> * ran e2e tests to verify existing tests are not affected by code changes.

@supershal can you please add the link to the tests so we have it for posterity

**Review by dkoshkin — APPROVED** [review]

**Comment by supershal** [comment]
> > * ran e2e tests to verify existing tests are not affected by code changes.
> 
> @supershal can you please add the link to the tests so we have it for posterity
https://github.com/mesosphere/konvoy-image-builder/actions/runs/5580524593 
The GHA job status does not show previous run tests properly.

**Pull request merged**